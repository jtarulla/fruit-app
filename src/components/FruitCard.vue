<template>
    <div class="fruit-card">
        <div class="fruit-card--img">
            <img :src=fruit.image :alt=fruit.name />
        </div>
        <div class="fruit-card--title">
            <div class="fruit-card--title-text">
                {{fruit.name}}
            </div>
            <div
                class="fruit-card--title-delete"
                @click="removeItem(fruit)"
            >
                <icon-delete />
            </div>
        </div>
        <router-link :to="`fruit/${fruit.id}`">
            <div class="fruit-card--view-more">
                details
                <icon-arrow />
            </div>
        </router-link>
    </div>
</template>

<script>
import { mapActions } from 'vuex';
import iconArrow from '@/assets/icon-arrow.vue';
import iconDelete from '@/assets/icon-delete.vue';

export default {
  name: 'FruitCard',
  components: {
      iconArrow,
      iconDelete
  },
  props: {
    fruit: {
        type: Object,
        required: true
    }
  },
  methods: {
    ...mapActions(['deleteFruit']),
    removeItem(item) {
        if (window.confirm(`Do you really want to delete ${item.name} item?`)) {
            this.deleteFruit(item);
        }
    }
  }
}
</script>
