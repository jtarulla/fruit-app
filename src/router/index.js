import Vue from 'vue';
import VueRouter from 'vue-router';
import store from '@/store';
import Home from '@/views/HomePage.vue';
import Error from '@/views/ErrorPage.vue';

Vue.use(VueRouter)

export const routes = [
  {
    path: '/',
    name: 'Home',
    component: Home
  },
  {
    path: '/fruit/:id',
    name: 'Fruit',
    component: () => import(/* webpackChunkName: "fruit-detail" */ '../views/FruitPage.vue')
  },
  {
    path: '*',
    name: 'Error',
    component: Error
  }
]

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes
})

router.beforeEach(async (to, from, next) => {
  await store.restored;
  next();
});

export default router
