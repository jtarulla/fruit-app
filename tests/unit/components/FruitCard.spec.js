import Vue from 'vue'
import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils'
import VueRouter from 'vue-router'
import router from '@/router'
import FruitCard from '@/components/FruitCard.vue'

const localVue = createLocalVue()
localVue.use(VueRouter)

describe('FruitCard component', () => {
  const fruit = {
    color: '',
    description: '',
    expires: '',
    id: 1,
    image: '',
    isFruit: true,
    name: '',
    price: '',
    taste: ''
  }
  
  let wrapper = null
  
  beforeEach(async () => {
    await router.push({
      name: 'Fruit',
      params: {id: fruit.id}
    }).catch(() => ({}));

    
    wrapper = mount(FruitCard, {
      localVue,
      router,
      stubs: {
        RouterLink: RouterLinkStub
      },
      propsData: {fruit}
    })
  })

  afterEach(async() => {
    jest.resetModules();
    jest.resetAllMocks();
  })

  it('renders the component', () => {
    expect(wrapper.vm.$options.name).toMatch('FruitCard')
    expect(wrapper.html()).toMatchSnapshot();
  })

  it('processes valid props data', async () => {
    wrapper.setProps({
      fruit: {
        id: 4,
      }
    })

    await Vue.nextTick();

    expect(wrapper.vm.fruit.id).toBe(4);
  })

  it('links to the right URL', async () => {
    wrapper.setProps({
      fruit: {
        id: 2,
      }
    })

    await Vue.nextTick();

    expect(wrapper.findComponent(RouterLinkStub).props().to).toBe('fruit/2')
  })
})