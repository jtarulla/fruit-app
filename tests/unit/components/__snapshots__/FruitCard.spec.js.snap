// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FruitCard component renders the component 1`] = `
<div class="fruit-card">
  <div class="fruit-card--img"><img src="" alt=""></div>
  <div class="fruit-card--title">
    <div class="fruit-card--title-text">

    </div>
    <div class="fruit-card--title-delete"><svg data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 50 50" role="img" aria-labelledby="icon-icon-close-description" class="icon icon-icon-close">
        <title id="icon-icon-close-description">icon close</title>
        <path d="M41.75 5L25 21.75 8.25 5 5 8.25 21.75 25 5 41.75 8.25 45 25 28.25 41.75 45 45 41.75 28.25 25 45 8.25 41.75 5z" data-name="SVG/Close" class="cls-1"></path>
      </svg></div>
  </div> <a>
    <div class="fruit-card--view-more">
      details
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 11.67 19.8">
        <g id="Layer_2" data-name="Layer 2">
          <g id="Layer_1-2" data-name="Layer 1">
            <path d="M0,18,1.77,19.8l9.9-9.9L1.77,0,0,1.77,8.13,9.9Z"></path>
          </g>
        </g>
      </svg>
    </div>
  </a>
</div>
`;
